src/main/java/no/ssb/rawdata/avro/cloudstorage/GCSRawdataUtils.java

```
@@ -97,7 +97,7 @@ static long getOffsetOfLastBlock(BlobId blobId) {
     Stream<Blob> listTopicFiles(String bucketName, String topic) {
         Page<Blob> page = storage.list(bucketName, Storage.BlobListOption.prefix(topic + "/"));
         Stream<Blob> stream = StreamSupport.stream(page.iterateAll().spliterator(), false);
-        return stream.filter(blob -> !blob.isDirectory());
+        return stream.filter(blob -> !blob.isDirectory() && blob.getSize() > 0);
```

This will work, but is a strategy where the cause of the problem is silently ignored. The root cause is the fact that there exists an empty file here which should not happen, and at the very least someone should be notified about it in order to investigate whether all data is intact.

---

This is basically a guard against a bug that occurs earlier in the same codebase. I guess it boils down to whether the original issue is easy or hard to fix.

We could throw in some warning logs, at the cost of slight clutter in the code.